tools/src/tester/StoryParser.ts

```diff
@@ -0,0 +1,39 @@
+/*
+* Copyright OpenSearch Contributors
+* SPDX-License-Identifier: Apache-2.0
+*
+* The OpenSearch Contributors require contributions made to
+* this file be licensed under the Apache-2.0 license or a
+* compatible open source license.
+*/
+
+import _ from "lodash";
+import { ChapterRequest, Story } from "./types/story.types";
+import { ParsedChapter, ParsedChapterRequest, ParsedStory } from "./types/parsed_story.types";
+
+export default class StoryParser {
+  static parse(story: Story): ParsedStory {
+    return {
+      ...story,
+      chapters: this.#expand_chapters(story.chapters) as ParsedChapter[],
+      prologues: this.#expand_chapters(story.prologues),
+      epilogues: this.#expand_chapters(story.epilogues)
+    }
+  }
+
+  static #chapter_methods(methods: string[] | string): string[] {
+    return [...(Array.isArray(methods) ? methods : [methods])]
+  }
+
+  static #expand_chapters(chapters?: ChapterRequest[]): ParsedChapterRequest[] {
+    if (chapters === undefined) return []
+    return  _.flatMap(_.map(chapters, (chapter) => {
+      return _.map(this.#chapter_methods(chapter.method), (method) => {
+        return {
+          ...chapter,
+          method
+        }
+      })
```

Comment:
We should add a suffix with the the name of the method at the end of the chapter synopsis to tell the expanded chapters apart. We should also mention this in the doc that such a chapter will be expanded into more chapters so they know what to expect in the result screen.

Reply:
Added, the output will now include the verb if there are multiple.

```
PASSED  template.yaml (tests/default/_core/render/template.yaml)
    PASSED  PROLOGUES
        PASSED  POST /_scripts/movie_template
    PASSED  CHAPTERS
        PASSED  Render the movie template (request payload). [GET]
            PASSED  REQUEST BODY
            PASSED  RESPONSE STATUS
            PASSED  RESPONSE PAYLOAD BODY
            PASSED  RESPONSE PAYLOAD SCHEMA
            SKIPPED RESPONSE OUTPUT VALUES
        PASSED  Render the movie template (request payload). [POST]
            PASSED  REQUEST BODY
            PASSED  RESPONSE STATUS
            PASSED  RESPONSE PAYLOAD BODY
            PASSED  RESPONSE PAYLOAD SCHEMA
            SKIPPED RESPONSE OUTPUT VALUES
        PASSED  Render the movie template (path). [GET]
            PASSED  PARAMETERS
                PASSED  id
            PASSED  REQUEST BODY
            PASSED  RESPONSE STATUS
            PASSED  RESPONSE PAYLOAD BODY
            PASSED  RESPONSE PAYLOAD SCHEMA
            SKIPPED RESPONSE OUTPUT VALUES
        PASSED  Render the movie template (path). [POST]
            PASSED  PARAMETERS
                PASSED  id
            PASSED  REQUEST BODY
            PASSED  RESPONSE STATUS
            PASSED  RESPONSE PAYLOAD BODY
            PASSED  RESPONSE PAYLOAD SCHEMA
            SKIPPED RESPONSE OUTPUT VALUES
    PASSED  EPILOGUES
        PASSED  DELETE /_scripts/movie_template

```